src/main/java/com/linecorp/armeria/common/http/HttpObject.java

```diff
@@ -19,4 +19,12 @@
 /**
  * The common interface for HTTP/2 message objects, {@link HttpHeaders} and {@link HttpData}.
  */
-public interface HttpObject {}
+public interface HttpObject {
+
+    /**
+     * Gets whether the stream should be ended when writing this object. This can be useful for
+     * {@link HttpHeaders}-only responses or to more efficiently close the stream along with the last piece of
+     * {@link HttpData}.
```

Comment:
I think the description of this method is somewhat misleading. Even if `isEndOfStream()` returns `false`, it can be the last object in the stream because a user can publish an object without EOS set and then close the stream. This method's returning `false` doesn't necessarily mean the stream will not end.